### PR TITLE
Source set support improvements

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Eliminates the `SpinePlugin` class which was only responsible for providing factory methods that did not depend on the state of the class.
  * Configuration of `protobuf` plugin performed by `ProtocConfigurationPlugin` was extracted into a method object.
  * Task-related types were placed under `io.spine.tools.gradle.task` sub-package. Previously they were just under `gradle`.
  * `SourceSetName` data class was introduced to make working with source sets more type-safe.

### CODE REVIEW

File: plugin-base/src/main/java/io/spine/tools/gradle/task/GradleTask.java [hunk 1]
```diff
@@ -65,6 +66,18 @@ private GradleTask(Task task, TaskName name, Project project) {
...
+    /**
+     * Creates a builder for a new task.
+     *
+     * @param name
+     *          the name of the task
+     * @param action
+     *          the configuration action for the task
+     */
+    public static Builder newBuilder(TaskName name, Action<Task> action) {
+        return new Builder(name, action);
```
Comment (armiol) [inline plugin-base/src/main/java/io/spine/tools/gradle/task/GradleTask.java:78]: I would check the parameters for `null`s here.

**Review by armiol — APPROVED** [review]
LGTM with a single suggestion.

File: plugin-base/src/main/kotlin/io/spine/tools/gradle/SourceSetName.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package io.spine.tools.gradle
+
+import org.gradle.api.tasks.SourceSet.MAIN_SOURCE_SET_NAME
+import org.gradle.api.tasks.SourceSet.TEST_SOURCE_SET_NAME
+
+/**
+ * A name of a Gradle project source set.
+ */
+public data class SourceSetName(val value: String) {
+    init {
```
Comment (dmdashenkov) [inline plugin-base/src/main/kotlin/io/spine/tools/gradle/SourceSetName.kt:36]:
```suggestion
public data class SourceSetName(val value: String) {

    init {
```

**Review by dmdashenkov — COMMENTED** [review]
LGTM with a comment.

**Pull request merged**